### PR TITLE
perf: track active configs and render on selection

### PR DESCRIPTION
**Pull request by joaoviana**
Related to: [#11047](https://github.com/lightdash/lightdash/issues/11047)

### Description:

We currently generate the data for all configs when the user hits run query. 

This gives a better experience between selecting different chart types, but slows down the initial load. 

This PR tracks the active chart config and _marks it as active_ so next time the user selects that same type, it won't re-mount.

Demo

https://github.com/user-attachments/assets/293aa5f8-5837-45ec-af4d-9736e9aec1d5

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx [hunk 1]
```diff
@@ -126,6 +116,30 @@ export const ContentPanel: FC = () => {
...
+                NonNullable<ReturnType<typeof selectChartConfigByKind>>,
+                TableChartSqlConfig
+            > => !isTableChartSQLConfig(c),
+        );
+
+        return {
+            chartConfigs,
+            tableConfig,
+        };
+    });
```
Comment (joaoviana) [inline packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx:141]: I could make this into its own selector separately, but would have to be a separate PR

**Review by owlas — APPROVED** [review]
